### PR TITLE
Update mono to 4.8.1

### PR DESCRIPTION
**Pull request by pcj**
- Download updated version from github rather than google mirror.
- Unpack archive with pkgutil on osx.
- Update mono/nuget-binary to latest.

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by pcj** [comment]
This was precipitated by nuget api error that now seem to be occurring with the older version...

File: dotnet/csharp.bzl [hunk 1]
```diff
@@ -447,6 +447,7 @@ def _nuget_package_impl(repository_ctx,
     nuget,
     "install",
     "-Version", repository_ctx.attr.version,
+    #"-Verbosity", "detailed", # enable for debugging
```
Comment (zaphar) [inline dotnet/csharp.bzl]: Can you move this to be a regular comment at the top. I prefer to leave out commented code.

Alternatively, you can put in an issue to add add this to the command when a debugging flag is turned on.

File: dotnet/csharp.bzl [hunk 2]
```diff
@@ -538,22 +539,70 @@ csharp_autoconf = repository_rule(
...
+  if not pkgutil:
+    fail("pkgutil not found in PATH")
+  tar = repository_ctx.which("tar")
+  if not tar:
+    fail("tar not found in PATH")
+
+  # Download the package (353MB)
+  repository_ctx.download(
+    "https://download.mono-project.com/archive/4.8.1/macos-10-universal/MonoFramework-MDK-4.8.1.0.macos10.xamarin.universal.pkg",
+    #"http://localhost:2017/MonoFramework-MDK-4.8.1.0.macos10.xamarin.universal.pkg",
```
Comment (zaphar) [inline dotnet/csharp.bzl]: Remove this comment.

File: dotnet/csharp.bzl (hunk 1 above)
Comment (pcj) [inline dotnet/csharp.bzl]: Added `verbosity` attr to nuget package with default 'normal'.

File: dotnet/csharp.bzl (hunk 2 above)
Comment (pcj) [inline dotnet/csharp.bzl]: Removed.

File: examples/example_nuget/BUILD [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+csharp_binary(
+    name = "hello",
+    srcs = [
+        "Program.cs",
+    ],
+    deps = [
+        "//examples/example_lib:MyClass",
+        "@ndesk_options//:dylibs",
+    ],
+)
```
Comment (pcj) [inline examples/example_nuget/BUILD]: Figured it would be good to have a nuget example.

**Pull request merged**